ethash/src/compute.rs

```diff
@@ -139,13 +139,14 @@ pub fn quick_get_difficulty(header_hash: &H256, nonce: u64, mix_hash: &H256, pro
 
 			let hash_len = header_hash.len();
 			buf[..hash_len].copy_from_slice(header_hash);
-			buf[hash_len..hash_len + mem::size_of::<u64>()].copy_from_slice(&nonce.to_ne_bytes());
+			let end = hash_len + mem::size_of::<u64>();
+			buf[hash_len..end].copy_from_slice(&nonce.to_ne_bytes());
 
-			keccak_512::unchecked(buf.as_mut_ptr(), 64, buf.as_ptr(), 40);
+			keccak_512::inplace_range(&mut buf, 0..end);
 			buf[64..].copy_from_slice(mix_hash);
 
 			let mut hash = [0u8; 32];
-			keccak_256::unchecked(hash.as_mut_ptr(), hash.len(), buf.as_ptr(), buf.len());
+			keccak_256::write(&buf, &mut hash);
```

Comment:
this change introduced a substantial regression, nothing stands out in benches https://github.com/paritytech/parity-common/pull/372 though

it shouldn't be a blocker for merging this PR though, I'll work on bringing the perf back